src/views/Settings/General/IgnorePendingTransactions.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback } from 'react';
+import { useInputProps } from '../../../lib';
+import { ACTION_SETTINGS_SET_IGNORE_PENDING_TRANSACTIONS } from '../../../budget';
+import Setting from '../Setting';
+import { Props } from './Types';
+
+export default function IgnorePendingTransactionsSetting({
+  state: {
+    settings: { ignorePendingTransactions },
+  },
+  dispatch,
+}: Props) {
+  const { value, onChange } = useInputProps({
+    value: ignorePendingTransactions,
+    onChange: useCallback(
+      (payload: boolean) => {
+        dispatch({ type: ACTION_SETTINGS_SET_IGNORE_PENDING_TRANSACTIONS, payload });
+      },
+      [dispatch],
+    ),
+    validate: useCallback((ev) => {
+      if (ev.target.value === true || ev.target.value === false) {
+        return ev.target.value;
+      } else if (!(ev.target as any).checked !== undefined) {
+        return (ev.target as any).checked
+      } else {
+        throw new Error('Error');
+      }
+    }, []),
+  });
```

Comment:
The `useInputProps` helper is not required here, since the setting can never be in a "invalid" state.
This is rather used for text-inputs that might be empty or have an invalid value. In that case They'll remember the latest valid value and thereby never create set of settings that might break the app.

I'd go with

```ts
const onChange = useCallback(
  (ev: React.ChangeEvent<HTMLInputElement>) => {
    dispatch({
      type: ACTION_SETTINGS_SET_IGNORE_PENDING_TRANSACTIONS,
      payload: ev.target.checked,
    });
  },
  [dispatch],
);

return (
  <Setting
    label="Ignore pending transactions"
    id="setting-ignore-pending-transactions"
  >
    <input
      type="checkbox"
      id="setting-ignore-pending-transactions"
      checked={ignorePendingTransactions}
      onChange={onChange}
    />
  </Setting>
);
```
  
To explain why this was not working: `ueInputProps` defaults to use `ev.target.value` as `value` which is `on` or `off` for checkboxes. (both are truthy in an if-statement). And the helper is currently not build to use different properties of the target or convert it internally.